src/relations.jl

```diff
@@ -1290,24 +1290,26 @@ is a function that is 1 above `T_freeze` and goes to zero below `T_icenuc`.
 function liquid_fraction(
     param_set::APS,
     T::FT,
-    ::Type{phase_type},
+    ::Type{State},
     q::PhasePartition{FT} = q_pt_0(FT),
-) where {FT <: Real, phase_type <: ThermodynamicState}
-    _T_freeze::FT = TP.T_freeze(param_set)
-    _T_icenuc::FT = TP.T_icenuc(param_set)
-
-    if T > _T_freeze
-        return FT(1)
-    elseif T > _T_icenuc
-        _pow_icenuc::FT = TP.pow_icenuc(param_set)
-        if _pow_icenuc == 1
-            return (T - _T_icenuc) / (_T_freeze - _T_icenuc)
-        else
-            return ((T - _T_icenuc) / (_T_freeze - _T_icenuc))^_pow_icenuc
-        end
-    else
-        return FT(0)
-    end
+) where State <: ThermodynamicState
```

Comment:
```suggestion
) where {FT, State <: ThermodynamicState}
```

Reply:
Style guides usually recommend getting `FT` from an `eltype` rather than type parameter. Should we use that instead then?

Reply:
Hmm I see that we need it for `q_pt_0` so perhaps best to leave the type parameter

Reply:
That would be fine, but this enforces diagonalization